drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -121,98 +270,233 @@ local function assign_child_profile(device, child_ep)
+          else
+            component_map["main"] = ep
+          end
+          component_map_used = true
+        else -- Create child devices for non-main endpoints
+          if ep ~= main_endpoint then -- don't create a child device that maps to the main endpoint
+            local name = string.format("%s %d", device.label, num_server_eps)
+            local child_profile = assign_child_profile(device, ep)
+            driver:try_create_device(
+                {
+                  type = "EDGE_CHILD",
+                  label = name,
+                  profile = child_profile,
+                  parent_device_id = device.id,
+                  parent_assigned_child_key = string.format("%d", ep),
+                  vendor_provided_label = name
+                }
+            )
+            current_component_number = current_component_number + 1
+          end
+        end
       end
     end
-  end
 
-  if num_server_eps > 1  then
-    -- If the device is a parent child device, then set the find_child function on init.
-    -- This is persisted because initialize switch is only run once, but find_child function should be set
-    -- on each driver init.
-    device:set_field(IS_PARENT_CHILD_DEVICE, true, {persist = true})
-  end
-
-  device:set_field(SWITCH_INITIALIZED, true)
-  -- The case where num_server_eps > 0 is a workaround for devices that have a
-  -- Light Switch device type but implement the On Off cluster as server (which is against the spec
-  -- for this device type). By default, we do not support Light Switch device types because by spec these
-  -- devices need bindings to work correctly (On/Off cluster is client in this case), so these device types
-  -- do not have a generic fingerprint and will join as a matter-thing. However, we have seen some devices
-  -- claim to be Light Switch device types and still implement their clusters as server, so this is a
-  -- workaround for those devices.
-  if num_server_eps > 0 and detect_matter_thing(device) == true then
-    local id = 0
-    for _, ep in ipairs(device.endpoints) do
-      -- main_endpoint only supports server cluster by definition of get_endpoints()
-      if main_endpoint == ep.endpoint_id then
-        for _, dt in ipairs(ep.device_types) do
-          -- no device type that is not in the switch subset should be considered.
-          if (ON_OFF_SWITCH_ID <= dt.device_type_id and dt.device_type_id <= ON_OFF_COLOR_DIMMER_SWITCH_ID) then
-            id = math.max(id, dt.device_type_id)
+    if num_server_eps > 1  then
+      -- If the device is a parent child device, then set the find_child function on init.
+      -- This is persisted because initialize switch is only run once, but find_child function should be set
+      -- on each driver init.
+      device:set_field(IS_PARENT_CHILD_DEVICE, true, {persist = true})
+    end
+
+    device:set_field(SWITCH_INITIALIZED, true)
+
+    if #button_eps > 0 then
```

Comment:
What if the device has a mixture of buttons and switch device types? For example, a button with an LED light on it? I think we need to determine how that kind of device type should be represented

Reply:
My thinking for this would be that if there are any button endpoints, the device will use an MCD configuration, and you can see [here](https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/blob/217b899ba18265f1e6123416e352a63343dc61b5/drivers/SmartThings/matter-switch/src/init.lua#L363) that a switch endpoint would use switch%d for its component name. I created an example profile of a 2 button, 1 switch device that would use this configuration. [This line](https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/blob/217b899ba18265f1e6123416e352a63343dc61b5/drivers/SmartThings/matter-switch/src/init.lua#L420) would append "-switch" to the profile name if there are switch endpoints in addition to button endpoints. If devices come out that need multiple switches or other configurations, we could update this profile selection logic and create new profiles for those devices, like we decided to do in the thermostat and sensor drivers. Do you think this approach would make sense?